client/components/date-range/inputs.js

```diff
@@ -0,0 +1,81 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+
+class DateRangeInputs extends Component {
+	static propTypes = {
+		startDateValue: PropTypes.string.isRequired,
+		endDateValue: PropTypes.string.isRequired,
+		onInputChange: PropTypes.func,
+		onInputBlur: PropTypes.func,
+		startLabel: PropTypes.string,
+		endLabel: PropTypes.string,
+	};
+
+	static defaultProps = {
+		startLabel: 'From',
+		endLabel: 'To',
+		onInputChange: noop,
+		onInputBlur: noop,
+	};
+
+	/**
+	 * Handles input events with fixed arguments
+	 * for consistency via partial application
+	 *
+	 * @param  {string} eventType  one of "change" or "blur"
+	 * @param  {string} startOrEnd one of "Start" or "End"
+	 * @return {function}            the partially applied function ready to recevie event data
+	 */
+	handleInputEvent = ( eventType, startOrEnd ) => e => {
+		const val = e.target.value;
+
+		if ( eventType === 'change' ) {
+			this.props.onInputChange( val, startOrEnd );
+		}
+
+		if ( eventType === 'blur' ) {
+			this.props.onInputBlur( val, startOrEnd );
+		}
+	};
+
+	render() {
+		return (
+			<div className="date-range__date-inputs">
+				<div className="date-range__date-input date-range__date-input--from">
+					<FormLabel htmlFor="startDate">{ this.props.startLabel }</FormLabel>
+					<FormTextInput
+						id="startDate"
+						name="startDate"
```

Comment:
Great catch. It could cause a problem. For some reason I thought react would prefix it in virtual dom but clearly I was way off!

Thinking of using https://lodash.com/docs/4.17.11#uniqueId to generate per instance Ids.

Reply:
I've utilised `uuid` to generate a per instance ID which I use to generate unique `input` IDs. 

![screen shot 2019-01-14 at 09 54 55](https://user-images.githubusercontent.com/444434/51106093-b8ae7700-17e2-11e9-8983-fdb65ebd516a.png)